pkg/auth/dependency/newinteraction/nodes/select_identity_oauth.go

```diff
@@ -2,46 +2,122 @@ package nodes
 
 import (
 	"errors"
+	"net/http"
 
 	"github.com/authgear/authgear-server/pkg/auth/config"
 	"github.com/authgear/authgear-server/pkg/auth/dependency/identity"
 	"github.com/authgear/authgear-server/pkg/auth/dependency/newinteraction"
 	"github.com/authgear/authgear-server/pkg/auth/dependency/sso"
 	"github.com/authgear/authgear-server/pkg/core/authn"
+	"github.com/authgear/authgear-server/pkg/core/crypto"
+	"github.com/authgear/authgear-server/pkg/core/skyerr"
 )
 
 func init() {
-	newinteraction.RegisterNode(&NodeSelectIdentityOAuth{})
+	newinteraction.RegisterNode(&NodeSelectIdentityOAuthProvider{})
+	newinteraction.RegisterNode(&NodeSelectIdentityOAuthUserInfo{})
 }
 
-type InputSelectIdentityOAuth interface {
-	GetUserInfo() sso.AuthInfo
+var ErrOAuthProviderNotFound = skyerr.NotFound.WithReason("OAuthProviderNotFound").New("oauth provider not found")
+
+type InputSelectIdentityOAuthProvider interface {
+	GetProviderAlias() string
```

Comment:
I remember you suggested to bundle these fields into a plain struct `FooArgs`, do we discard that idea now?

Reply:
The struct does not seem to bring any values. Interface with multiple methods should be fine.